internal/processor/filtermetric/config.go

```diff
@@ -16,13 +16,31 @@ package filtermetric
 
 import (
 	"go.opentelemetry.io/collector/internal/processor/filterset"
+	"go.opentelemetry.io/collector/internal/processor/filterset/regexp"
+)
+
+// MatchType specifies the strategy for matching against `pdata.Metric`s. This
+// is distinct from filterset.MatchType which matches against metric (and
+// tracing) names only. To support matching against metric names and
+// `pdata.Metric`s, filtermetric.MatchType is effectively a superset of
+// filterset.MatchType.
+type MatchType string
+
+// These are the MatchTypes that users can specify for filtering
+// `pdata.Metric`s.
+const (
+	Regexp MatchType = MatchType(filterset.Regexp)
+	Strict MatchType = MatchType(filterset.Strict)
+	Expr   MatchType = "expr"
 )
 
 // MatchProperties specifies the set of properties in a metric to match against and the
 // type of string pattern matching to use.
 type MatchProperties struct {
-	// MatchConfig configures the matching patterns used when matching metric properties.
-	filterset.Config `mapstructure:",squash"`
```

Comment:
This inlines the previously-squashed `filterset.Config` so that we can use the additional `expr` match_type that doesn't belong in the `filterset` package (`filterset` is just for string matching).